tests/js/server/aql/aql-queries-geo.js

```diff
@@ -1358,7 +1358,36 @@ function geoLegacyComparison() {
                                  RETURN d._key`, {}, false);
       assertEqual(0, b.length);
     },
+  };
+}
+
+function geoThisAndThatSuite() {
+
+  let coll;
 
+  const cn = "UnitTestsAhuacatlGeoThisAndThat";
+
+  return {
+    setUp : function() {
+      db._drop(cn);
+      coll = db._create(cn);
+      coll.ensureIndex({type:"geo", fields:["geo"], geoJson: true});
+    },
+
+    tearDown : function() {
+      db._drop(cn);
+    },
+
+    testNestedFor : function() {
+      let query = `FOR e IN ${cn}
+                     FOR d IN ${cn}
+                       FILTER GEO_INTERSECTS(d.geo, e.geo)
+                       RETURN {d: d._key, e: e._key}`;
+      let compact = helper.getCompactPlan(AQL_EXPLAIN(query))
+        .map(function(node) { return node.type; });
+      assertEqual([ "SingletonNode", "EnumerateCollectionNode",
```

Comment:
IMO we should at least check here that the selected index is the expected geo index, and not for example the primary index.

Reply:
Indeed, done.